libvast/include/vast/fbs/flatbuffer_container.hpp

```diff
@@ -0,0 +1,88 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2022 The VAST Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#pragma once
+
+#include "vast/fwd.hpp"
+
+#include "vast/chunk.hpp"
+#include "vast/fbs/segmented_file.hpp"
+
+namespace vast::fbs {
+
+/// This container provides a `std::vector<vast::chunk>`-like
+/// interface for chunks that were created from
+/// a `flatbuffer_container_builder`.
+// The typical usage is for the first chunk to be some flatbuffer
+// and the subsequent ones being data blobs that are too big to
+// store inline without hitting the 2GiB limit for individual
+// flatbuffers.
+class flatbuffer_container {
+public:
+  // The `chunk` must begin with a flatbuffer that has
+  // `fbs::SegmentedFileHeader` as root type.
+  explicit flatbuffer_container(vast::chunk_ptr chunk);
+
+  /// Get the chunk at position `idx`.
+  chunk_ptr get_raw(size_t idx) const;
+
+  /// Return the chunk at position `idx` interpreted as a `T`.
+  template <typename T>
```

Comment:
```suggestion
  template <typename T>
    requires (std::is_trivial_v<T>)
```

Otherwise the below reinterpret_cast would be UB, as you'd need `std::start_lifetime_as` which afaict is not yet implemented in all compilers we support.